jetty-util/src/main/java/org/eclipse/jetty/util/thread/ReservedThreadExecutor.java

```diff
@@ -62,13 +69,14 @@ public String toString()
 
     private final Executor _executor;
     private final int _capacity;
-    private final ConcurrentLinkedDeque<ReservedThread> _stack;
-    private final AtomicInteger _size = new AtomicInteger();
-    private final AtomicInteger _pending = new AtomicInteger();
+    private final Set<ReservedThread> _threads = ConcurrentHashMap.newKeySet();
+    private final SynchronousQueue<Runnable> _queue = new SynchronousQueue<>(false);
+    private final AtomicBiInteger _count = new AtomicBiInteger(); // hi=pending; lo=size;
 
     private ThreadPoolBudget.Lease _lease;
     private long _idleTime = 1L;
     private TimeUnit _idleTimeUnit = TimeUnit.MINUTES;
```

Comment:
I would convert these 2 fields into 1 (in nanos), makes it so much easier to use it in the rest of the code.

Reply:
I made it ms as that is a better resolution.